src/Form/WebPageArchiveQueueForm.php

```diff
@@ -49,51 +50,90 @@ public static function create(ContainerInterface $container) {
   /**
    * {@inheritdoc}
    */
-  public function getFormId() {
-    return 'web_page_archive_queue_form';
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function buildForm(array $form, FormStateInterface $form_state) {
-    $queue = $this->queueFactory->get('web_page_archive_capture');
-
-    $form['help'] = [
-      '#type' => 'markup',
-      '#markup' => $this->t('Submitting this form will process the web page archive queue which contains @number items.', ['@number' => $queue->numberOfItems()]),
-    ];
-    $form['actions']['#type'] = 'actions';
-    $form['actions']['submit'] = [
-      '#type' => 'submit',
-      '#value' => $this->t('Process queue'),
-      '#button_type' => 'primary',
-    ];
+  public function form(array $form, FormStateInterface $form_state) {
+    $form = parent::form($form, $form_state);
+    $web_page_archive = $this->getEntity();
+    $this->queueCt = $web_page_archive->getQueueCt();
+    // TODO: Provide better instructions for these forms.
+    if ($this->queueCt > 0) {
+      $form['help'] = [
+        '#type' => 'markup',
+        '#markup' => $this->t('Submitting this form will process the web page archive queue which contains @number items.', ['@number' => $web_page_archive->getQueueCt()]),
+      ];
+    }
+    else {
+      $form['help'] = [
+        '#type' => 'markup',
+        '#markup' => $this->t('Click to start.'),
+      ];
+    }
 
     return $form;
   }
 
   /**
-   * {@inheritdoc}
+   * Processes the queue.
    */
-  public function submitForm(array &$form, FormStateInterface $form_state) {
-    $queue = $this->queueFactory->get('web_page_archive_capture');
-    $queue_worker = $this->queueManager->createInstance('web_page_archive_capture');
+  public function processQueue(array $form, FormStateInterface $form_state) {
+    // TODO: Move this behavior to controller.
+    // TODO: Add batch processing.
+    $web_page_archive = $this->getEntity();
+    $queue = $this->queueFactory->get("web_page_archive_capture.{$web_page_archive->uuid()}");
+    $queue_worker = $this->queueManager->createInstance("web_page_archive_capture");
 
+    $processed_ct = 0;
     while ($item = $queue->claimItem()) {
       try {
         $queue_worker->processItem($item->data);
         $queue->deleteItem($item);
+        $processed_ct++;
       }
       catch (SuspendQueueException $e) {
         $queue->releaseItem($item);
+        watchdog_exception('web_page_archive', $e);
```

Comment:
It could be too DRY but the identical call in multiple places with scalar values could be wrapped in a single local method that facilitates testing and refactors. Not by any means a hard request, just a suggestion for something to look at.